BLL/Services/CityService.cs

```diff
@@ -0,0 +1,31 @@
+﻿using BLL.Interfaces;
+using DAL.Repositories;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace BLL.Services
+{
+    public class CityService : ICityService
+    {
+        private protected IUnitOfWork _db;
+
+        public CityService(IUnitOfWork db)
+        {
+            _db = db;
+        }
+
+        public string GetLocationById(int id)
+        {
+            var city = _db.Cities.Get(id).CityName;
```

Comment:
too many calls to the DB...
you can retrieve the city object and then use its properties like id, name and etc
the same issue with the states